templates/sources.lists.j2

```diff
@@ -0,0 +1,75 @@
+# {{ ansible_managed }}
+{%- macro write_entry(suite, release_path) %}
+{{- fmtstr|format('deb', mirror, suite, release, release_path, components) }}
+{%- if gen_sources %}
+	{{- fmtstr|format('deb-src', mirror, suite, release, release_path, components) }}
+{%- endif %}
+{%- endmacro %}
+
+{%- macro write_additional_entries() %}
+{% if additional_source_strings|length >= 1 %}
+
+## Additional repositories
+{% for repo in additional_source_strings %}
+{{ repo }}
+{% endfor %}
+{% endif %}
+{%- endmacro %}
+
+
+{%- if ansible_distribution == 'Debian' %}
+
+{% set mirror = apt_sources_debian.mirror|default(apt_sources_debian_mirror) %}
+{% set release = apt_sources_debian.release|default(ansible_distribution_release) %}
+{% set components = apt_sources_debian.components|default(apt_sources_debian_components)|join(' ') %}
+{% set gen_sources = apt_sources_debian.enable_sources|default(apt_sources_debian_enable_sources) %}
+{% set additional_source_strings = apt_sources_debian.additional_source_strings|default(apt_sources_debian_additional_source_strings) %}
+{%- if gen_sources %}
```

Comment:
Wouldn't it be possible to check this value in the macro and adapt `fmtstr` accordingly there? That way, you wouldn't have this duplicated.

Reply:
Sadly not, I first tried at that way.